src/mercury_engine_data_structures/formats/bmsld.py

```diff
@@ -220,5 +277,52 @@ def compare_func(first: str, second: str) -> bool:
 
     def insert_into_entity_group(self, sub_area: Container, name_to_add: str) -> None:
         # MSR requires to have the names in the sub area list sorted by their crc32 value
-        sub_area.names.append(name_to_add)
-        sub_area.names.sort(key=crc32)
+        sub_area.objects.append(name_to_add)
+        sub_area.objects.sort(key=crc32)
+
+    def _get_layer(self, layer: ActorLayer) -> ListContainer:
+        """Returns a layer of actors"""
+        return self.raw.actor_layers[layer]
+
+    def _check_if_actor_exists(self, layer: ActorLayer, actor_name: str) -> None:
+        if actor_name not in self._get_layer(layer):
+            raise KeyError(f"No actor named '{actor_name}' found in '{layer}!'")
+
+    def get_actor(self, layer: ActorLayer, actor_name: str) -> BmsldActor:
+        """Returns an actor given a layer and an actor name"""
+        self._check_if_actor_exists(layer, actor_name)
+        return BmsldActor(self.raw.actor_layers[layer][actor_name])
+
+    def remove_actor(self, layer: ActorLayer, actor_name: str) -> None:
+        """Deletes an actor given a layer and an actor name"""
+        self._check_if_actor_exists(layer, actor_name)
+        self._get_layer(layer).pop(actor_name)
+        self.remove_actor_from_all_groups(actor_name)
+
+    def copy_actor(
+        self,
+        position: list[float],
+        template_actor: BmsldActor,
+        new_name: str,
+        layer: ActorLayer,
+        offset: tuple = (0, 0, 0),
+    ) -> BmsldActor:
+        """
+        Copies an actor to a new position
+
+        param position: the x, y, z position for the copied actor
+        param template_actor: the actor being copied
+        param new_name: the name for the copied actor
+        param layer: the layer the copied actor will be added to
+        param offset: adds an additional offset the copied actor's coordinates
+
+        """
+        new_actor = BmsldActor(copy.deepcopy(template_actor))
+        self.raw.actor_layers[layer][new_name] = new_actor._raw
+        new_actor.position = Vec3([p + o for p, o in zip(position, offset)])
+
+        return new_actor
+
+    def get_logic_shape(self, logic_shape: str) -> Container:
```

Comment:
looks like we'll need to add a class for collision objects soon too. I'm ok if that's in another pr and this function returns a container for now, just add a TODO comment

Reply:
Doesn't this already exist?

Reply:
I mean if it already exists then by all means use it lol